### DISCUSSION TITLE
Persistent Storage (for survival maps etc...)?

### DISCUSSION THREAD

**Discussion by TropicLegend** [jeremylvln/Shulker#249] [Q&A] [answered]
Is there a way to add persistent storage (for survival maps etc...)?

**Comment by TropicLegend** [comment]
Maybe there might also be a way for a sidecar to be attached that just copies/backs up the world data.

**Comment by TropicLegend** [comment]
I have found a solution using the podOverrides.
I was able to add custom volumeMounts, enabling me to add volume claims to the minecraft-server pod.

**Reply by TropicLegend** [reply]
Okay,  it seems it was not the correct answer as it indeed did not work.
For some reason, the volumes are not mounted into the pod even though I do not get an error when starting up the pod.

**Reply by TropicLegend** [reply]
Do you maybe know a possible solution @jeremylvln ?
It  seems they are do not attach to the pod that is started.

**Reply by jeremylvln** [reply, marked as answer]
I looked at the code and the `podOverrides` does not support adding volumes, but it can be something to consider.
Agones is by design not meant for persistence, but only for ephemeral workload. I knew this use case would come one day or another (but was hoping to not deal with it this soon haha).

I guess I can still add the option to attach volumes, and it will be your job to deal with that :)

But please note that if you want to persist a map, you'll not be able to use `MinecraftServerFleet`s because you'll not be able to share your volume with multiple replicas, use `MinecraftServer`s directly

**Reply by TropicLegend** [reply]
Thank you very much! 
I am looking forward to that option to attach volumes as it would make things much easier :+1:
